.github/workflows/merge.yml

```diff
@@ -179,24 +179,59 @@ jobs:
           SN_LOG: "all"
         timeout-minutes: 2
 
-      - name: Start a client to create a register
+      - name: Start a client to create a register writable only by the owner
         run: cargo run --bin safe --release -- --log-output-dest=data-dir register create -n baobao
         env:
           SN_LOG: "all"
         timeout-minutes: 10
 
-      - name: Start a client to get a register
+      - name: Start a client to get a register writable only by the owner (client is the owner)
         run: cargo run --bin safe --release -- --log-output-dest=data-dir register get -n baobao
         env:
           SN_LOG: "all"
         timeout-minutes: 2
 
-      - name: Start a client to edit a register
+      - name: Start a client to edit a register writable only by the owner (client is the owner)
         run: cargo run --bin safe --release -- --log-output-dest=data-dir register edit -n baobao wood
         env:
           SN_LOG: "all"
         timeout-minutes: 10
 
+      - name: Start a client to create a register writable by anyone
+        run: cargo run --bin safe --release -- --log-output-dest=data-dir register create -n trycatch -p
+        env:
+          SN_LOG: "all"
+        timeout-minutes: 10
+
+      - name: Start a client to get a register writable by anyone (client is the owner)
+        run: cargo run --bin safe --release -- --log-output-dest=data-dir register get -n trycatch
+        env:
+          SN_LOG: "all"
+        timeout-minutes: 2
+
+      - name: Start a client to edit a register writable by anyone (client is the owner)
+        run: cargo run --bin safe --release -- --log-output-dest=data-dir register edit -n trycatch water
+        env:
+          SN_LOG: "all"
+        timeout-minutes: 10
+
+      - name: Delete client subdir to generate new client
+        shell: bash
+        run: rm -rf "${{ matrix.safe_path }}"/client
+
+      - name: Start a client to get a register writable by anyone (new client, the owner of Register)
+        run: cargo run --bin safe --release -- --log-output-dest=data-dir register get -n trycatch
+        env:
+          SN_LOG: "all"
+        timeout-minutes: 2
+
+      - name: Start a client to edit a register writable by anyone (new client, not the owner of Register)
+        run: cargo run --bin safe --release -- --log-output-dest=data-dir register edit -n trycatch water
```

Comment:
edit the register to a new value other than `water`

also, better have another step of `use the new client to edit the previous privately owned register (baobao)`